### PR TITLE
Support addition of license header to generated files

### PR DESCRIPTION
**Pull request by cstamas**
Add ability to set license header (verbatim text of file) and push that into generated files.

### CODE REVIEW

File: modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java [hunk 1]
```diff
@@ -160,11 +173,29 @@ public void execute() throws MojoExecutionException {
...
             parameters.setProperty(
                     ModelloParameterConstants.ALL_VERSIONS, StringUtils.join(packagedVersions.iterator(), ","));
         }
 
+        if (licenseText != null && !licenseText.trim().isEmpty()) {
+            String license = "";
+            try {
+                Path licenseFile = Paths.get(licenseText);
+                if (Files.isRegularFile(licenseFile)) {
+                    license = String.join("\n", Files.readAllLines(licenseFile));
```
Comment (slawekjaranowski) [inline modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:186]: Should we managed in special way for line-end char?
Should be default or system specific?

Comment (cstamas) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:186]: I just copied from other (existing) code that adds "generated" header. Unsure why would it need to differ.

File: modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java [hunk 2]
```diff
@@ -95,6 +99,15 @@ public abstract class AbstractModelloGeneratorMojo extends AbstractMojo {
...
     private List<String> packagedVersions = new ArrayList<String>();
 
+    /**
+     * The contents of license header, verbatim. It may be file path and if file exists, will be loaded up. Otherwise,
+     * the contents of this parameter is reused as-is.
+     *
+     * @since 2.3.1
+     */
+    @Parameter
+    private String licenseText;
```
Comment (slawekjaranowski) [inline modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:109]: I would like to consider two parameters:
 - one with plain text
 - one with file name 

Code will be simple and we don't need a magic to discover what content is

Comment (cstamas) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:109]: Agreed, so mojo have two params.... model config has one that holds string (that is loaded from file or was param). My bigger problem is how header is mutilated by XML writer... newlines are getting removed and so on.

Comment (cstamas) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:109]: Also, if two params... what happens if both are configured?

Comment (slawekjaranowski) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:105]: someone wins ... or throw an error

Comment (cstamas) [reply modello-maven-plugin/src/main/java/org/codehaus/modello/maven/AbstractModelloGeneratorMojo.java:105]: I documented: licenseText config "prevails" if both set. That's all.

File: modello-maven-plugin/src/it/maven-model/pom.xml [hunk 3]
```diff
@@ -28,12 +28,14 @@
           <models>
             <model>src/main/mdo/maven.mdo</model>
           </models>
+          <licenseText>The license of this file</licenseText>
```
Comment (slawekjaranowski) [inline modello-maven-plugin/src/it/maven-model/pom.xml:31]: simple assertions will be appreciated if text present in output

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**